### PR TITLE
Bug 1771335: Verify pullsecret builds

### PR DESCRIPTION
**Pull request by adambkaplan**

### CODE REVIEW

**Comment by adambkaplan** [comment]
fyi @gabemontero @nalind - this should verify that we're using a pull secret correctly in builds. Right now _something_ is preventing this from working correctly.

**Comment by adambkaplan** [comment]
@gabemontero squashed this, gcp-builds should pass with your fix.

**Comment by gabemontero** [comment]
google iam policy problem with e2e-gcp-builds

**Comment by gabemontero** [comment]
more ci install flakes with e2e-gcp-builds

**Comment by gabemontero** [comment]
still install flakes on e2e-gcp-builds

**Comment by gabemontero** [comment]
ok we got a clean install but your test case still failed @adambkaplan 

apparently my manual testing using my tbr cred and the new-app invocation from the bug is not quite the same as your new test case

I'll start running your new test case manually with some additional debug and see what is going on

File: test/extended/testdata/builds/pullsecret/pullsecret-nodejs-bc.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  source:
+    git:
+      uri: "https://github.com/sclorg/nodejs-ex.git"
+  strategy:
+    type: Source
+    sourceStrategy:
+      from:
+        kind: DockerImage
+        name: registry.redhat.io/rhscl/nodejs-10-rhel7:latest
+    pullSecret:
```
Comment (gabemontero) [inline test/extended/testdata/builds/pullsecret/pullsecret-nodejs-bc.yaml]: @adambkaplan looks like the indentation on you yaml is resulting in this pull secret not getting pulled in

See that it is missing when I use it as is to create the BC:  
```
gmontero ~/QE_bzs/img-src-auth $ cat pullsecret-nodejs.yaml 
kind: BuildConfig
apiVersion: v1
metadata:
  name: pullsecret-nodejs
spec:
  source:
    git:
      uri: "https://github.com/sclorg/nodejs-ex.git"
  strategy:
    type: Source
    sourceStrategy:
      from:
        kind: DockerImage
        name: registry.redhat.io/rhscl/nodejs-10-rhel7:latest
    pullSecret:
      name: local-ps
gmontero ~/QE_bzs/img-src-auth $ oc get bc -o yaml
apiVersion: v1
items:
- apiVersion: build.openshift.io/v1
  kind: BuildConfig
  metadata:
    creationTimestamp: "2019-11-15T18:35:13Z"
    name: pullsecret-nodejs
    namespace: ggmtest
    resourceVersion: "49618"
    selfLink: /apis/build.openshift.io/v1/namespaces/ggmtest/buildconfigs/pullsecret-nodejs
    uid: e4c6808a-da49-4a11-b9e1-ca736e533014
  spec:
    failedBuildsHistoryLimit: 5
    nodeSelector: null
    output: {}
    postCommit: {}
    resources: {}
    runPolicy: Serial
    source:
      git:
        uri: https://github.com/sclorg/nodejs-ex.git
      type: Git
    strategy:
      sourceStrategy:
        from:
          kind: DockerImage
          name: registry.redhat.io/rhscl/nodejs-10-rhel7:latest
      type: Source
    successfulBuildsHistoryLimit: 5
    triggers: []
  status:
    lastVersion: 1
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
gmontero ~/QE_bzs/img-src-auth $
```
when I indent `pullSecret` in by 2 spaces, it is incorporated:

```
gmontero ~/QE_bzs/img-src-auth $ oc get bc -o yaml
apiVersion: v1
items:
- apiVersion: build.openshift.io/v1
  kind: BuildConfig
  metadata:
    creationTimestamp: "2019-11-15T18:45:03Z"
    name: pullsecret-nodejs
    namespace: ggmtest
    resourceVersion: "52110"
    selfLink: /apis/build.openshift.io/v1/namespaces/ggmtest/buildconfigs/pullsecret-nodejs
    uid: e56ff415-94da-4da0-b20b-a53f2e7c49dd
  spec:
    failedBuildsHistoryLimit: 5
    nodeSelector: null
    output: {}
    postCommit: {}
    resources: {}
    runPolicy: Serial
    source:
      git:
        uri: https://github.com/sclorg/nodejs-ex.git
      type: Git
    strategy:
      sourceStrategy:
        from:
          kind: DockerImage
          name: registry.redhat.io/rhscl/nodejs-10-rhel7:latest
        pullSecret:
          name: local-ps
      type: Source
    successfulBuildsHistoryLimit: 5
    triggers: []
  status:
    lastVersion: 0
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
gmontero ~/QE_bzs/img-src-auth $ 
```

and the build ultimatey works with my changes to openshift/builder

please update this and we'll try the e2e in this PR again


```

Comment (gabemontero) [reply test/extended/testdata/builds/pullsecret/pullsecret-nodejs-bc.yaml]: the corrected pullsecret-nodejs yaml:

```
gmontero ~/QE_bzs/img-src-auth $ cat pullsecret-nodejs.yaml
kind: BuildConfig
apiVersion: v1
metadata:
  name: pullsecret-nodejs
spec:
  source:
    git:
      uri: "https://github.com/sclorg/nodejs-ex.git"
  strategy:
    type: Source
    sourceStrategy:
      from:
        kind: DockerImage
        name: registry.redhat.io/rhscl/nodejs-10-rhel7:latest
      pullSecret:
        name: local-ps
gmontero ~/QE_bzs/img-src-auth $ 
```

Comment (adambkaplan) [reply test/extended/testdata/builds/pullsecret/pullsecret-nodejs-bc.yaml]: @gabemontero gotta love YAML. Fixed in the next push.

Comment (gabemontero) [reply test/extended/testdata/builds/pullsecret/pullsecret-nodejs-bc.yaml]: yes you do :-)

**Comment by gabemontero** [comment]
prometheus failures in e2e-gcp

**Comment by gabemontero** [comment]
e2e-gcp-builds is passing @adambkaplan

**Comment by gabemontero** [comment]
install flake e2e-gcp

/retest

**Comment by gabemontero** [comment]
@adambkaplan do we want to cherrypick this test back to 4.2 / 4.1 since we are backporting the CVE fix to not always setting skip tls verify to true to those destinations ?

**Pull request merged**